app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -369,15 +443,135 @@ export default {
     // ---------------------------------------------------------------------------------------------------
     init() {
       // Since change is triggered during loading
-      this.setDirtyFlag(false);
     },
-    onChangeMethod() {
+    onChangeMethod(_changed, _flags, _modified) {
```

Comment:
`onChangeMethod` is a bit ambiguous as to what event it is handling, and for what object/context. I would propose creating another method called something like `onSchemaChange` as it makes it more clear the intent of this method (do something with the schema when it changes), and  then calling that function within the `onChangeMethod` function. IIRC `onChangeMethod` is supposed to mirror/override the internal form.io event hooks, so it would be safer to functionalize out our history changes into its own invokable method instead.